src/index.jsx

```diff
@@ -31,35 +42,71 @@ function getConfig() {
 function bootstrap() {
   return new Promise((resolve, reject) => {
     // We need to do this (in order) before bothering to render:
-    // 1. grab the configuration + load Google's auth2 library (in parallel)
-    // 2. get the client id out of the config and use it to logIn
-    // 3. put the config and user details into the store
-    // 4. render
-    getConfig()
-      .then((configData) => {
+    // 1. Check status of the server
+    // 2. grab the configuration + load Google's auth2 library (in parallel)
+    // 3. get the client id out of the config and use it to logIn
+    // 4. put the config and user details into the store
+    // 5. render
+    checkStatus()
+      .then((response) => {
         store.dispatch({
-          type: ActionTypes.GET_CONFIGURATION_SUCCESS,
-          configuration: configData,
+          type: ActionTypes.GET_SERVER_STATUS_SUCCESS,
+          status: {
+            tdrOperational: true,
+            apiIsUp: true,
+            serverStatus: response.data,
+          },
         });
-        getUser({ client_id: configData.clientId })
-          .then((user) => {
-            if (user != null) {
-              store.dispatch(
-                logIn(
-                  user.name,
-                  user.imageUrl,
-                  user.email,
-                  user.accessToken,
-                  user.accessTokenExpiration,
-                ),
-              );
-              store.dispatch(getFeatures());
-            }
-            resolve();
+        getConfig()
+          .then((configData) => {
+            store.dispatch({
+              type: ActionTypes.GET_CONFIGURATION_SUCCESS,
+              configuration: configData,
+            });
+            getUser({ client_id: configData.clientId })
+              .then((user) => {
+                if (user != null) {
+                  store.dispatch(
+                    logIn(
+                      user.name,
+                      user.imageUrl,
+                      user.email,
+                      user.accessToken,
+                      user.accessTokenExpiration,
+                    ),
+                  );
+                  store.dispatch(getFeatures());
+                }
+                resolve();
+              })
+              .catch(reject);
           })
           .catch(reject);
+        resolve();
       })
-      .catch(reject);
+      .catch((error) => {
+        console.error('The server is down!');
+        if (error.response.status === 504) {
+          store.dispatch({
+            type: ActionTypes.GET_SERVER_STATUS_DOWN,
+            status: {
+              tdrOperational: false,
+              apiIsUp: false,
+              serverStatus: error.response.data,
+            },
+          });
+        } else {
+          store.dispatch({
+            type: ActionTypes.GET_SERVER_STATUS_FAILURE,
+            status: {
+              tdrOperational: false,
+              apiIsUp: true,
```

Comment:
I'm wondering what situation this would happen in. I thought if there was any failure status from the API, this would be false.

Reply:
If the TDR api is up, but SAM is not, then the API is running but TDR isn't operational. We can use that information to give a more detailed message about whats going on vs just "the server is down".

Reply:
Hmmm so I'm looking at the [status endpoint code](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/configuration/StatusService.java#L46), and the Data Repo API status is only "ok" if SAM and Postgres are up.

Reply:
As discussed at stand-up, the first condition occurs when the api pod is unavailable (so there is no response at all), while this one occurs if we can reach the status endpoint, but there's a 503 response.